File: pairSum.js
const given = [10, 15, 3, 7];
const wanted = 17;
console.log(findPairSum(given, wanted));

function findPairSum(array, number) {
  let map = new Map();
  for (let x = 0, length = array.length; x < length; x++) {
    if (map.has(array[x])) {
      return true;
    } else {
      map.set(number - array[x], array[x]);
    }
  }
  return false;
}

File: pair_sum.py
def find_pair_sum(array, number):
    dict = {}
    for x in range(len(array)):
        if array[x] in dict:
            return True
        else:
            dict[number - array[x]] = array[x]

    return False


given = [10, 15, 3, 7]
wanted = 17
print(find_pair_sum(given, wanted))
